src/InlineInput/NumberInputWithDropdown.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React, { useState } from 'react';
+import FormLabel from 'react-bootstrap/esm/FormLabel';
+
+import { DropdownItem } from '../Dropdown/Dropdown';
+import classNames from '../utils/classNames';
+import NumberInlineInput from './NumberInlineInput';
+
+import styles from '../Dropdown/Dropdown.module.scss';
+
+export type NumberDropdownItem = DropdownItem<number>;
+
+interface DropdownProps {
+    id?: string;
+    label?: React.ReactNode;
+    items: NumberDropdownItem[];
+    value: number;
+    onChange: (value: number) => void;
+    numItemsBeforeScroll?: number;
+    className?: string;
+    disabled?: boolean;
+}
+
+export default ({
+    id,
+    label,
+    items,
+    value,
+    onChange,
+    numItemsBeforeScroll = 0,
+    className = '',
+    disabled = false,
+}: DropdownProps) => {
+    const [isActive, setIsActive] = useState(false);
+    const [inlineValue, setInlineValue] = useState(value);
+
+    const setNewValue = (newValue: number) => {
+        setInlineValue(newValue);
+        onChange(newValue);
+        setIsActive(false);
+    };
+    const onClickItem = (item: NumberDropdownItem) => {
+        if (item.value != null && typeof item.value === 'number') {
+            setNewValue(item.value);
+        }
+    };
+
+    return (
+        <div
+            className={`tw-preflight tw-relative tw-w-full ${className}`}
+            onBlur={event => {
+                if (!event.currentTarget.contains(event.relatedTarget)) {
+                    setIsActive(false);
+                }
+            }}
+        >
+            <FormLabel className="tw-mb-1 tw-text-xs">{label}</FormLabel>
+            <div className="tw-flex tw-h-8 tw-w-full">
+                <NumberInlineInput
+                    value={inlineValue}
+                    range={{ min: 1, max: 1_000_000, step: 1, decimals: 0 }}
```

Comment:
This is wrong, should be a property to the component and then passed to the `NumberInlineInput`